lib/reek/ast/sexp_extensions/module.rb

```diff
@@ -45,28 +45,45 @@ module ModuleNode
       # Utility methods for :class nodes.
       module ClassNode
         include ModuleNodeBase
+
         def superclass() children[1] end
+        def ancestor() superclass end
       end
 
       # Utility methods for :casgn nodes.
       module CasgnNode
         include ModuleNodeBase
 
         def defines_module?
-          return false unless value
-          call = case value.type
-                 when :block
-                   value.call
-                 when :send
-                   value
-                 end
+          call = module_definition
           call && call.module_creation_call?
         end
 
+        def module_definition
```

Comment:
I *think* this method returns more than just any module creation calls. For example, it would return a value for the code

```
Foo = foo(42)
```

Can we think of a better name?